cli/azd/pkg/tools/github/github_test.go

```diff
@@ -142,6 +142,45 @@ func TestNewGitHubCli(t *testing.T) {
 	require.Equal(t, Version.String(), ver)
 }
 
+func TestGetAuthStatus(t *testing.T) {
```

Comment:
If we feel that manually testing all the conditions is too tedious when we upgrade `gh`, we could create a set of contracts tests that tests against the "real live" gh and run againast all the known contracts as inputs and assert the expected azd behaviors.

Just want to highlight that this test will only catch that azd is working optimally in this code flow, but will not catch contractual changes if gh changes their outputs again.

Reply:
I was thinking how to create such contracts, but running a real live gh cli will have some challenges to control its state when running on CI/CD.  We might not want to logout when gh cli is logged in automatically, and things like that.

So, having at least a little coverage for now was my starting point.

Since we use the stdout/stderr from many tools as contracts, we might want to think on how to protect ourself from breaking changes like this one, but without manipulating the state around the tool too much